### ISSUE TITLE
Formula Input

### ISSUE THREAD

**Issue by Suunami** [zombieFox/awesomeSheet#171]
This might be something you're already considering, since I notice certain things like Skills and CMB having dropdown checkboxes for auto-calculating. Something like this would be very cool for automating attack and damage bonuses.

Example, instead of simply putting +11/6 for your sixth level fighter, you could input =bab+str+2 and it'll automatically pull the Bab of 6 and Str modifier of 3, then add 2. This way, when a character's bonuses change, other aspects of the sheet will follow suit. To save headaches with extra attacks, this formula method could be applied to each individual attack.

On that note, a section of powers that can be toggled would be neat. Put in ability score modifiers and you can have a Rage toggle, automated Wild Shape, etc. Not sure how robust this feature could be without bogging the whole sheet down though.

Of course, this all sounds like a whole lot of work on your part, but I figured I'd throw it out there anyways. Cheers.

**Comment by zombieFox** [comment]
Hi, yes I've given this much thought over the course of the project. Individual attacks having auto calculating attack bonuses would be very handy. There are however lots of underlying concerns about how to achieve this. The acceptance criteria for such a feature would be something like:

- the application allowing a user:
  - to configure attack x with as either "Melee" or "Ranged"
  - to add temporary modifiers, eg: Haste +1 to all attacks
    - to configure which attack receive those temporary modifiers
  - to configure bonuses from Feats for any number of attacks
  - to configure bonuses from Class Features for any number of attacks
  - to mark a power as "Active"
    - to configure which attack receive those power modifiers
- the application should automatically:
  - update all attacks configured as "Melee" or "Ranged" when the corresponding stat changes
 
That being said I've already begun down this road with features like the multiple Class BAB calculations. The other aspects of the above (and I'm sure there are more scenarios than listed) need a lot of development to implement correctly. I also feel that it would be a poor user experience if the above were only partially implemented.

In short, I'm working on building the underlying requirements to achieve the solution you described. 
(Side note, apologies for the late reply, life is quite busy right now.)
